crates/vm/levm/src/vm.rs

```diff
@@ -144,12 +147,12 @@ impl VM {
                     env.origin,
                     new_contract_address,
                     new_contract_address,
-                    code,
+                    code.clone(),
                     value,
-                    Bytes::new(),
+                    code,
```

Comment:
Why this?

Reply:
Because I think the calldata field should have the actual calldata, and `code` in the case of create is the calldata.
Not doing this I guess will result in undesired behavior when calling for example `CALLDATACOPY` in a `Create` transaction.
Maybe I can replace code for `calldata.clone()` to make it more clear

Reply:
https://github.com/lambdaclass/ethrex/pull/1345/commits/74162a5e337ea82f25b7dbafe49b367af5b7309d
I added this if it helps